src/core/geometry/qgsgeos.cpp

```diff
@@ -512,6 +512,65 @@ double QgsGeos::hausdorffDistanceDensify( const QgsAbstractGeometry *geom, doubl
   return distance;
 }
 
+double QgsGeos::frechetDistance( const QgsAbstractGeometry *geom, QString *errorMsg ) const
+{
+#if GEOS_VERSION_MAJOR==3 && GEOS_VERSION_MINOR<9
```

Comment:
```suggestion
#if GEOS_VERSION_MAJOR==3 && GEOS_VERSION_MINOR<7
```
(according to https://postgis.net/docs/manual-dev/ST_FrechetDistance.html and tthe doc of qgsgeos.h)